File: E-ConstatBackEnd/src/main/java/com/example/ecanstatbackend/entity/VehiculeInformation.java
package com.example.ecanstatbackend.entity;

import lombok.*;

import javax.persistence.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Entity
public class VehiculeInformation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @OneToOne(cascade = CascadeType.ALL)
    private InsurenceCompanyInformation insurenceCompanyInformation;
    @OneToOne(cascade = CascadeType.ALL)
    private DriverInformation driverInformation;
    @OneToOne(cascade = CascadeType.ALL)
    private InsuredInforamation insuredInforamation;
    @OneToOne(cascade = CascadeType.ALL)
    private Vehicule vehicule;
    private String apperantDamage;
    private String observation;
    private String signature;

}

File: E-ConstatFrontEnd/src/app/model/vehicule-information.ts
import { DriverInformation } from "./driver-information";
import { InsurenceCompanyInformation } from "./insurance-company-information";
import { InsuredInformation } from "./insured-inforamation";
import { Vehicule } from "./vehicule";

export class VehiculeInformation{
    id:                          number;
    insurenceCompanyInformation: InsurenceCompanyInformation;
    driverInformation:           DriverInformation;
    insuredInforamation:         InsuredInformation;
    vehicule:                    Vehicule;
    apperantDamage:              string;
    observation:                 string;
    signature:                   string;
    circumstances: string;
}